samples/nrf9160/lwm2m_client/src/main.c

```diff
@@ -92,14 +91,31 @@ static int remove_whitespace(char *buf)
 	return 0;
 }
 
+/* remove trailing AT OK message */
+static void remove_ok(char *buf)
+{
+	remove_whitespace(buf);
+
+	uint8_t *ok_resp = strstr(imei_buf, "OK");
+	uint32_t index = ok_resp - imei_buf;
```

Comment:
Is it guaranteed that `AT+CGSN` will return `OK` response? I.e. can it fail? I think it'd be safer to check if `ok_resp` is not a NULL pointer at this point.

Reply:
Trimming of the string happens after calling `query_modem` which already checks the AT response from the modem and would fail early if met with a return value different from 0. When return value is 0, the buffer will contain the imei and the OK string, which we need to trim.

Reply:
It that case we should verify the result of `query_modem()` call before calling `remove_ok()`, no?